apps/05Mastermind.js

```diff
@@ -15,29 +15,73 @@ function printBoard() {
     }
 }
 
-function generateSolution() {
+function generateSolution(solution, guess) {
     for (var i = 0; i < 4; i++) {
         var randomIndex = getRandomInt(0, letters.length);
         solution += letters[randomIndex];
     }
 }
 
 function getRandomInt(min, max) {
-  return Math.floor(Math.random() * (max - min)) + min;
+    return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
     // your code here
+
+    var solutionArray = solution.split('');
+    var guessArray = guess.split('');
+
+    var correctLetterLocations = 0;
+    var correctLetters = 0
+
+    for (var i = 0; i < solutionArray.length; i++) {
+        if (solutionArray[i] == guessArray[i]) {
+            correctLetterLocations++;
+            solutionArray[i] == null;
+        }
+    }
+
+    for (var i = 0; i < solutionArray.length; i++) {
+        for (var j = 0; j < guessArray.length; j++) {
+            if (solutionArray[i] == guessArray[j]) {
+                correctLetters++;
+            }
+
+        }
+
+    }
+
+    return correctLetterLocations.toString().red + '-' + correctLetters.toString().white + '-';
 }
+console.log(generateHint("abcd", "abdc"));
+
+
 
 function mastermind(guess) {
     // your code here
+    if (guess == solution) {
+        return 'You guessed it!';
+    } else if (board.length == 10) {
+        return 'You ran out of turns! The solution was ' + solution;
+        process.exit();
```

Comment:
Return statements cause the function to stop. This means any code after a return statement will never be run, process.exit() in this case.